### PR TITLE
Remove Module::IsManifest and IsPEFile

### PR DESCRIPTION
**Pull request by elinor-fung**
- `Module::IsManifest`: should always be true now that we don't support multi-module.
- `Module::IsPEFile`: checks that the PE assembly is not dynamic - just doing that check seemed clearer to me than the `IsPEFile` name (especially now that we no longer have a `PEFile` class)

cc @lambdageek

### CODE REVIEW

File: src/coreclr/debug/ee/debugger.cpp [hunk 1]
```diff
@@ -9393,53 +9393,11 @@ void Debugger::LoadModule(Module* pRuntimeModule,
...
         return;
 
-    // If this is a dynamic module, then it's part of a multi-module assembly. The manifest
-    // module within the assembly contains metadata for all the module names in the assembly.
-    // When a new dynamic module is created, the manifest module's metadata is updated to
-    // include the new module (see code:Assembly.CreateDynamicModule).
-    // So we need to update the RS's copy of the metadata. One place the manifest module's
-    // metadata gets used is in code:DacDbiInterfaceImpl.GetModuleSimpleName
-    //
-    // See code:ReflectionModule.CaptureModuleMetaDataToMemory for why we send the metadata-refresh here.
```
Comment (jkotas) [inline src/coreclr/debug/ee/debugger.cpp:9403]: Delete "manifest module is updated with the name of a new dynamic module." from the comment on CaptureModuleMetaDataToMemory

File: src/coreclr/debug/daccess/request.cpp [hunk 2]
```diff
@@ -1744,7 +1744,7 @@ ClrDataAccess::GetModuleData(CLRDATA_ADDRESS addr, struct DacpModuleData *Module
     EX_TRY
     {
         //
-        // In minidump's case, these data structure is not avaiable.
+        // In minidump's case, these data structure is not available.
```
Comment (jkotas) [inline src/coreclr/debug/daccess/request.cpp:1747]:
```suggestion
        // In minidump's case, these data structures are not available.
```

**Review by jkotas — APPROVED** [review]
Thanks!

File: src/coreclr/vm/eventtrace.cpp [hunk 3]
```diff
@@ -4321,7 +4321,7 @@ VOID ETW::LoaderLog::SendModuleEvent(Module *pModule, DWORD dwEventOptions, BOOL
 
     PCWSTR szDtraceOutput1=W(""),szDtraceOutput2=W("");
     BOOL bIsDynamicAssembly = pModule->GetAssembly()->IsDynamic();
-    BOOL bIsManifestModule = pModule->IsManifest();
+    BOOL bIsManifestModule = TRUE; // Always the manifest module - multi-module assemblies are no longer supported
```
Comment (am11) [inline src/coreclr/vm/eventtrace.cpp:4324]: nit: this has only one usage, can be deleted?

File: src/coreclr/debug/daccess/request.cpp [hunk 4]
```diff
@@ -1733,7 +1733,7 @@ ClrDataAccess::GetModuleData(CLRDATA_ADDRESS addr, struct DacpModuleData *Module
     ModuleData->metadataSize = (SIZE_T) metadataSize;
 
     ModuleData->bIsReflection = pModule->IsReflection();
-    ModuleData->bIsPEFile = pModule->IsPEFile();
+    ModuleData->bIsPEFile = !pModule->GetPEAssembly()->IsDynamic();
```
Comment (lambdageek) [inline src/coreclr/debug/daccess/request.cpp:1736]: would it make sense to add a module kind enum to `Module` and replace both `Module::IsPEFile()` and `Module::IsReflection()`?

Alternately, can this be `!pModule->IsReflection()`?

Comment (jkotas) [reply src/coreclr/debug/daccess/request.cpp:1736]: `IsReflectionEmit` would be probably the best self-describing name for `IsReflection`, `IsDynamic`, `!IsPEFile` - all these mean the same thing.

**Review by lambdageek — APPROVED** [review]

**Pull request merged**